util/patricia_trie/src/lib.rs

```diff
@@ -49,155 +49,159 @@ mod lookup;
 mod nibbleslice;
 mod nibblevec;
 
-pub use self::triedbmut::TrieDBMut;
 pub use self::triedb::{TrieDB, TrieDBIterator};
+pub use self::triedbmut::TrieDBMut;
 pub use self::sectriedbmut::SecTrieDBMut;
 pub use self::sectriedb::SecTrieDB;
 pub use self::fatdb::{FatDB, FatDBIterator};
 pub use self::fatdbmut::FatDBMut;
 pub use self::recorder::Recorder;
+use node_codec::NodeCodec;
+
+// REVIEW: is this overdoing it?
+pub type KeccakRlpNodeCodec = node_codec::RlpNodeCodec<hashdb::KeccakHasher>;
+pub type KeccakTrieResult<T> = Result<T, <KeccakHasher as Hasher>::Out>;
 
 /// Trie Errors.
 ///
 /// These borrow the data within them to avoid excessive copying on every
 /// trie operation.
 #[derive(Debug, PartialEq, Eq, Clone)]
-pub enum TrieError {
+pub enum TrieError<T> {
 	/// Attempted to create a trie with a state root not in the DB.
-	InvalidStateRoot(H256),
+	InvalidStateRoot(T),
 	/// Trie item not found in the database,
-	IncompleteDatabase(H256),
+	IncompleteDatabase(T),
 	/// Corrupt Trie item
-	DecoderError(rlp::DecoderError),
+	// REVIEW: what we'd really like to do here is include the `rlp::DecoderError` in the `TrieError`
```

Comment:
what we'd specifically want to do is define `TrieError<T, E>` where `E=rlp::DecoderError` in Ethereum-like chains.

Reply:
(and provide a typedef either here or in `patricia-trie-ethereum` if we go that route, `type TrieError = generic::TrieError<H256, rlp::DecoderError>`)